substrate/frame/staking-async/src/pallet/impls.rs

```diff
@@ -880,6 +885,92 @@ impl<T: Config> Pallet<T> {
 		Eras::<T>::pending_rewards(era, &account)
 	}
+
+	/// Calculates the total stake of the lowest portion validators and stores it for the planned
+	/// era.
+	pub(crate) fn calculate_lowest_total_stake(era: EraIndex) {
+		// Only calculate if unbonding queue params have been set.
+		if let Some(params) = UnbondingQueueParams::<T>::get() {
+			// Determine the total stake from the lowest portion of validators and persist for the
+			// era.
+			let mut total_stake: Vec<_> =
+				ElectableStashes::<T>::get().into_iter().map(|(_, b)| b).collect();
+			let validators_to_check =
+				(params.lowest_ratio * total_stake.len() as u32).max(1) as usize;
+
+			// Sort exposure total stake by the lowest first and truncate to the lowest portion.
+			total_stake.sort();
+			total_stake.truncate(validators_to_check);
+
+			// Calculate the total stake of the lowest portion validators.
+			let total_stake = total_stake.into_iter().sum();
+
+			// Store the total stake of the lowest portion validators for the planned era.
+			Eras::<T>::set_lowest_stake(era, total_stake)
+		}
+	}
+
+	/// Curates the unlocking chunks for a stash account by:
+	/// - Calculating immediately withdrawable balance.
+	/// - Determining new unlock eras based on queue parameters.
+	/// - Organizing remaining chunks by era.
+	///
+	/// Returns a tuple of:
+	/// - Amount that can be immediately withdrawn.
+	/// - Map of era to remaining unlock chunks.
+	pub(crate) fn curate_unlocking_chunks(
+		stash: T::AccountId,
+	) -> ((EraIndex, BalanceOf<T>), BTreeMap<EraIndex, UnlockChunk<BalanceOf<T>>>) {
+		let current_era = CurrentEra::<T>::get().unwrap_or(Zero::zero());
+		let ledger = match Self::ledger(Stash(stash)) {
+			Ok(l) => l,
+			Err(_) => return ((current_era, Zero::zero()), BTreeMap::new()),
+		};
+		let target_era = current_era.saturating_add(1).saturating_sub(T::BondingDuration::get());
+		let (min_unlock_era, min_slashable_share) = match UnbondingQueueParams::<T>::get() {
+			None => (T::BondingDuration::get(), Zero::zero()),
+			Some(params) => (params.unbond_period_lower_bound, params.min_slashable_share),
+		};
+		let mut result = BTreeMap::new();
+		let mut free = BalanceOf::<T>::zero();
+		for chunk in ledger.unlocking.into_iter() {
+			let max_release_era = chunk.era.defensive_saturating_add(T::BondingDuration::get());
+			if current_era >= max_release_era {
+				// We can immediately withdraw these funds.
+				free.saturating_accrue(chunk.value);
+			} else {
+				let mut final_era = chunk.era.defensive_saturating_add(min_unlock_era);
+				for era in (target_era..=chunk.era).rev() {
+					let mut total_unbond = BalanceOf::<T>::zero();
+					let era_total_amount =
+						TotalUnbondInEra::<T>::get(chunk.era).unwrap_or_default();
+					let unbond = if era == chunk.era {
+						era_total_amount.min(
+							chunk.previous_unbonded_stake.defensive_saturating_add(chunk.value),
+						)
+					} else {
+						era_total_amount
+					};
+					total_unbond.saturating_accrue(unbond);
```

Comment:
`total_unbond` is always same as `unbond` since you are reinitialising `total_unbond` in every iteration.

Reply:
Indeed. As you mentioned, `total_unbond` is meant to accumulate over several iterations. That was fixed in previous commits.